src/components/Class/ClassList/index.js

```diff
@@ -83,94 +95,159 @@ function ClassList({ editClass, isShow }) {
+
+  const pathwayFilter = canSpecifyFacilitator
+    ? classData.filter((item) => {
+        return item?.pathway_id === pathwayID;
+      })
+    : classData;
   return (
     <>
-      <TextField
-        size={isActive ? "normal" : "small"}
-        variant="outlined"
-        InputProps={{
-          startAdornment: (
-            <InputAdornment position="start">{<SearchIcon />}</InputAdornment>
-          ),
-        }}
-        placeholder="Enter Batch or Class Name"
-        value={filterText}
-        sx={{
-          margin: isActive ? "0 0 0 4px" : "12px 0 0 2px",
-          width: isActive ? "98%" : "99%",
-          borderRadius: "8px",
-        }}
-        onPaste={(e) => {
-          e.preventDefault();
-          setFilterText(e.clipboardData.getData("text"));
+      {data && data.length > 0 && (
+        <Box
+          display={!isActive && "flex"}
+          sx={{
+            justifyContent: !isActive && "space-between",
+            marginTop: "32px",
+          }}
+        >
+          <TextField
+            size={"small"}
+            variant="outlined"
+            InputProps={{
+              style: { paddingTop: "3px" },
+              startAdornment: (
+                <InputAdornment position="start">
+                  {<SearchIcon />}
+                </InputAdornment>
+              ),
+            }}
+            placeholder="Enter Batch or Class Name"
+            value={filterText}
+            sx={{
+              margin: isActive ? "0 0 0 4px" : "0px 0 0 0px",
+              maxWidth: isActive ? "100%" : "99%",
+              borderRadius: "8px",
+            }}
+            fullWidth={isActive && true}
+            onPaste={(e) => {
+              e.preventDefault();
+              setFilterText(e.clipboardData.getData("text"));
 
-          let filtered_recurring_classes = recurring_classes.filter(
-            (item) =>
-              item.title
-                .toLowerCase()
-                .indexOf(e.clipboardData.getData("text").toLowerCase()) > -1
-          );
+              let filtered_recurring_classes = recurring_classes.filter(
+                (item) =>
+                  item.title
+                    .toLowerCase()
+                    .indexOf(e.clipboardData.getData("text").toLowerCase()) > -1
+              );
 
-          set_recurring_classes_data_set(filtered_recurring_classes);
-        }}
-        onChange={(e) => {
-          setFilterText(e.target.value);
-          if (filterText?.length > 0) {
-            let filtered_recurring_classes = recurring_classes.filter(
-              (item) =>
-                item.title.toLowerCase().indexOf(e.target.value.toLowerCase()) >
-                -1
-            );
-            set_recurring_classes_data_set(filtered_recurring_classes);
-          } else {
-            set_recurring_classes_data_set(null);
-          }
-        }}
-      />
+              set_recurring_classes_data_set(filtered_recurring_classes);
```

Comment:
Use `const` instead of `let`. Also, you can define a function with this code, so you don't need to repeat it as you do below on lines 149 - 155.